plugins/techdocs/src/home/components/TechDocsCustomHome.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { useAsync } from 'react-use';
+import { makeStyles } from '@material-ui/core';
+
+import { catalogApiRef, CatalogApi } from '@backstage/plugin-catalog-react';
+import { Entity } from '@backstage/catalog-model';
+import {
+  CodeSnippet,
+  Content,
+  ConfigApi,
+  configApiRef,
+  Header,
+  HeaderTabs,
+  Page,
+  Progress,
+  useApi,
+  WarningPanel,
+  SupportButton,
+  ContentHeader,
+} from '@backstage/core';
+import { DocsTable } from './DocsTable';
+import { DocsCardGrid } from './DocsCardGrid';
+
+const widgets = {
+  DocsTable: DocsTable,
+  DocsCardGrid: DocsCardGrid,
+};
+
+export type WidgetType = 'DocsCardGrid' | 'DocsTable';
+
+export interface WidgetConfig {
+  title: string;
+  description: string;
+  widgetType: WidgetType;
+  widgetMaxHeight?: string;
```

Comment:
This seems like a curious inclusion to enshrine in a config interface.  Is there no better alternative?

Reply:
@iamEAP I was hoping to allow stacking of widgets vertically within a tab but that would mean that the vertical height of a widget had to be determined somehow. For DocsCardGrid it might also be a full tab view so always restricting the vertical height would not really work.

Perhaps a `widgetOptions` object that had different options in it based on what widget was being used? And widgetMaxHeight is moved into there. We could export something like DocsCardGridOptionsType or DocsTableOptionsType as a definition for the options. 

I'm open to other suggestions, not entirely sure the best approach.

Reply:
Hmm....  Yeah, splitting out different options for different types seems reasonable, although that feels like it could get verbose.

A couple of other thoughts that came to mind:
- Just allowing styles to be passed down directly on a new property in the config?  Perhaps similar to how it's handled within `<ItemCardGrid />` [itself](https://github.com/backstage/backstage/blob/master/packages/core/src/layout/ItemCard/ItemCardGrid.tsx#L32).
- Introducing some kind of a grid system to the page and then exposing column/row properties on the widget configs?

...But the further I go down those lines of thinking, the more it feels like just a poor man's config-driven template system.  ...And maybe what we actually want is just a template system where one can mix and match the composable elements.

Reply:
I suggest we have this basic config system for the time being, and we can look at where to go next as we gather more feedback on the views users might want.

Reply:
I've added a 'panelCSS' optional object to pass CSS to the panel (minus its content header) to allow controlling panel height, amongst other things. Eg. `panelCSS: { maxHeight:400px }`.  Hopefully this allows some flexibility without making the configuration parameters more convoluted.

Reply:
Thanks!  That sounds good.

I wonder too if we could resolve the fixed height / stacking issue by applying `emptyRowsWhenPaging: false` to the underlying `<Table />` component in `<DocsTable />`?

Reply:
Will try that for <Table />, wasn't familiar

Reply:
It seems it's not straightforward to limit the `<Table />`'s height with an outside container, with or without `emptyRowsWhenPaging`. I think that the better UX is to have only one `<Table/>` per tab anyway, so I think we can leave it as is for the time being.

Reply:
updated for above